source/quick-start.txt

```diff
@@ -7,7 +7,122 @@ Quick Start
 .. contents:: On this page
    :local:
    :backlinks: none
-   :depth: 1
+   :depth: 2
    :class: singlecol
 
-Lorem ipsum
+.. include:: /includes/quick-start/overview.rst
+
+.. note::
+   If you are working with an existing application that uses the legacy API,
+   identified by depending on the ``mongodb-driver-legacy.jar`` library, read
+   the :doc:`Legacy Quick Start Guide </quick-start-legacy>` instead of this
+   one.
+
+Set up Your Project
+-------------------
+
+Install the Java Development Kit (JDK)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. include:: /includes/quick-start/install-jdk.rst
+
+Create the Project
+~~~~~~~~~~~~~~~~~~~
+
+.. include:: /includes/quick-start/create-project.rst
+
+Add MongoDB as a Dependency
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you are using Maven, add the following to your ``pom.xml`` dependencies
+list:
+
+.. code-block:: xml
+
+   <dependencies>
+       <dependency>
+           <groupId>org.mongodb</groupId>
+           <artifactId>mongodb-driver-sync</artifactId>
+           <version>4.0.1</version>
+       </dependency>
+   </dependencies>
+
+If you are using Gradle, add the following to your ``build.gradle``
+dependencies list:
+
+.. code-block:: groovy
+
+   dependencies {
+     compile 'org.mongodb:mongodb-driver-sync:4.0.1'
+   }
+
+.. note::
+   You may need to configure the JDK version in your IDE and/or dependency
+   file. Refer to the documentation provided by maintainers of your IDE
+   and/or dependency management system.
+
+   For example, if you build with Maven in the IntelliJ IDE, you need to
+   specify your JDK version explicitly in the ``properties`` section in your
+   ``pom.xml`` file. The following example instructs Maven to use JDK
+   version 1.9 (Java 9):
+
+   .. code-block:: xml
+
+      <properties>
+          <maven.compiler.source>1.9</maven.compiler.source>
```

Comment:
I'm not sure you should have to do this.  Will have to check it.

Reply:
It's not required in Eclipse, but IntelliJ IDEA has some strange behavior of defaulting to 1.5 if not specified as Nathan discovered.
https://youtrack.jetbrains.com/issue/IDEA-215058?_ga=2.114859969.797257447.1586200689-1373391973.1586200689

Reply:
I don't think it's enough just to set the property values though.  You would need to use them in a block like:

    <build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.8.1</version>
                <configuration>
                    <release>8</release>
                    <source>8</source>
                    <target>8</target>
                </configuration>
            </plugin>
        </plugins>
    </build>

Reply:
OK, I see it too.  Let's specify 1.8 though since it's the minimum version required by the driver and still most popular.